# World Poverty Measure

Data Analysis with Pandas

Team members:

Sandeep Yadav Pala Sreeramulu      (Net id: spalas3)
Thashma Nellira Dechamma           (Net id: dn7)
Akshat Gupta                       (Net id: akshatg2)

In [1]:
import re
import pandas as pd
import numpy as np

In [2]:
'''
This dataset contains the world poverty data based on the Multi Dimensional Poverty index which has the following 10 factors:
Health :- Nutrition, Child mortality
Education :- Years of schooling, School attendance
Living Standards :- Cooking fuel, Sanitation, Drinking water, Electricity, Housing, Assets
'''
xls = pd.ExcelFile('MPI Dataset 2021.xlsx')
mpi_df = pd.read_excel(xls, 'Sheet1')
mpi_df.sample(10)

Country Numeric Code Country Code      Country  \
74                   586          PAK     Pakistan   
6                    417          KGZ   Kyrgyzstan   
23                   662          LCA  Saint Lucia   
7                    400          JOR       Jordan   
73                   516          NAM      Namibia   
87                   686          SEN      Senegal   
39                   818          EGY        Egypt   
66                   320          GTM    Guatemala   
42                   484          MEX       Mexico   
92                    24          AGO       Angola   

                       World Region  Survey        Year  \
74                       South Asia      DHS  2017-2018   
6           Europe and Central Asia     MICS       2018   
23  Latin America and the Caribbean     MICS       2012   
7                       Arab States      DHS  2017-2018   
73               Sub-Saharan Africa      DHS       2013   
87               Sub-Saharan Africa      DHS       2019   
39                      Arab States      DHS       2014   
66  Latin America and the Caribbean      DHS  2014-2015   
42  Latin America and the Caribbean  ENSANUT       2016   
92               Sub-Saharan Africa      DHS  2015-2016   

    Multidimensional Poverty Index (MPI)  Nutrition  Child mortality  \
74                              0.198247  26.980925         5.871063   
6                               0.001426   0.393031         0.159708   
23                              0.007202   1.501055              NaN   
7                               0.001526   0.175637         0.167731   
73                              0.184735  31.336466         3.738906   
87                              0.262862  26.815081         5.828751   
39                              0.019682   3.912896         0.812845   
66                              0.133518  18.985577         2.103575   
42                              0.025615   5.233283              NaN   
92                              0.282435  29.269051         6.596673   

    Years of schooling  School attendance  Cooking fuel  Sanitation  \
74           24.819756          24.323150     31.241473   21.687021   
6             0.039468           0.113286      0.264173    0.052981   
23            0.325211           0.000000      0.325427    0.709397   
7             0.245264           0.244469      0.009403    0.039202   
73            7.330641           8.095706     38.360582   37.286172   
87           32.530376          43.878088     46.993851   28.860202   
39            2.864399           3.403865           NaN    0.706924   
66           15.974353          12.065980     27.843192   10.840516   
42            1.324077           0.778597      2.431594    2.161883   
92           31.397652          22.914641     39.797891   44.354364   

    Drinking water  Electricity    Housing     Assets  \
74        7.943032     7.142214  30.643726  12.203164   
6         0.096483     0.000000   0.036623   0.000000   
23        0.152276     0.307255   1.103478   0.383556   
7         0.057727     0.000000   0.107639   0.033383   
73       20.889700    36.656897  31.909369  15.914284   
87       15.726519    25.812155  22.526227   6.075707   
39        0.326330     0.053229   0.736992   0.214209   
66       10.474930    10.120378  22.739497  10.925110   
42        1.163711     0.139551   1.432236   1.070969   
92       36.907376    43.184752  44.128182  29.476472   

    Total number of indicators included  Indicator (s) missing  
74                                    10                   NaN  
6                                     10                   NaN  
23                                     9       Child mortality  
7                                     10                   NaN  
73                                    10                   NaN  
87                                    10                   NaN  
39                                     9          Cooking fuel  
66                                  

In [3]:
world_bank_df = pd.read_csv('World Bank Data.csv')
# The dataset from World Bank has the world poverty which is the traditional income based
world_bank_df = world_bank_df.sort_values(by = 'World_Bank_Estimate', ascending=False)
world_bank_df = world_bank_df.reset_index(drop=True)
world_bank_df

Country  Population_Below_Poverty_National_Year  \
0              Slovak Republic                                  2017.0   
1                      Morocco                                  2013.0   
2                    St. Lucia                                  2016.0   
3           West Bank and Gaza                                  2016.0   
4           Russian Federation                                  2018.0   
..                         ...                                     ...   
211                Korea, Rep.                                     NaN   
212                      Nauru                                     NaN   
213                New Zealand                                     NaN   
214   Northern Mariana Islands                                     NaN   
215  Sint Maarten (Dutch part)                                     NaN   

     Population_Below_Poverty_National  Population_Below_Poverty_Rural  \
0                                 12.2                             NaN   
1                                  4.8                             1.4   
2                                 25.0                            12.2   
3                                 29.2                             0.8   
4                                 12.9                             NaN   
..                                 ...                             ...   
211                                NaN                             NaN   
212                                NaN                             NaN   
213                                NaN                            39.1   
214                                NaN                             4.0   
215                                NaN                            25.1   

     Population_Below_Poverty_Urban  Poverty_Gap_Year  Reference_Year  \
0                               NaN               NaN          2014.0   
1                              15.0            2017.0          2015.0   
2                              44.0            2014.0          1999.0   
3                               4.5            2016.0          2015.0   
4                               NaN               NaN          2017.0   
..                              ...               ...             ...   
211                             NaN               NaN             NaN   
212                             NaN               NaN             NaN   
213                            71.0            2018.0             NaN   
214                            35.1            2015.0             NaN   
215                            58.8            2013.0             NaN   

     World_Bank_Estimate  Income_Lowest_10  Income_Lowest_20  ...  \
0                   63.0               0.9               2.4  ...   
1                   59.1               1.0               2.8  ...   
2                   57.6               NaN               NaN  ...   
3                   57.1               1.0               2.9  ...   
4                   56.3               1.3               3.9  ...   
..                   ...               ...               ...  ...   
211                  NaN               NaN               NaN  ...   
212                  NaN               NaN               NaN  ...   
213                  NaN               NaN               NaN  ...   
214                  NaN               NaN               NaN  ...   
215                  NaN               NaN               NaN  ...   

     Income_Highest_20  Income_Highest_10  Baseline_Year  Recent_Year  \
0                 68.2               50.5            NaN          NaN   
1                 63.7               47.3            NaN          NaN   
2                 60.2               42.7            NaN          NaN   
3                 61.3               44.4         2011.0       2016.0   
4                 61.2               49.1            NaN          NaN   
..                 ...                ...            ...          ...   
211                NaN      

In [4]:
#read the data from OPHI
national_mpi_df = pd.read_csv('MPI_national.csv')
national_mpi_df

ISO                   Country  MPI Urban  Headcount Ratio Urban  \
0    KAZ                Kazakhstan      0.000                    0.0   
1    SRB                    Serbia      0.000                    0.1   
2    KGZ                Kyrgyzstan      0.000                    0.1   
3    TUN                   Tunisia      0.000                    0.1   
4    ARM                   Armenia      0.001                    0.2   
..   ...                       ...        ...                    ...   
97   CAF  Central African Republic      0.289                   58.2   
98   LBR                   Liberia      0.290                   60.5   
99   SOM                   Somalia      0.293                   55.9   
100  TCD                      Chad      0.351                   64.8   
101  SSD               South Sudan      0.459                   82.5   

     Intensity of Deprivation Urban  MPI Rural  Headcount Ratio Rural  \
0                              33.3      0.000                   0.09   
1                              41.4      0.002                   0.50   
2                              40.2      0.003                   0.70   
3                              35.6      0.012                   3.18   
4                              33.3      0.001                   0.39   
..                              ...        ...                    ...   
97                             49.7      0.519                  89.79   
98                             48.0      0.481                  84.86   
99                             52.4      0.651                  96.92   
100                            54.1      0.609                  93.41   
101                            55.7      0.591                  94.00   

     Intensity of Deprivation Rural  
0                              33.3  
1                              40.3  
2                              37.1  
3                              38.7  
4                              36.9  
..                              ...  
97                             57.8  
98                             56.6  
99                             67.2  
100                            65.2  
101                            62.8  

[102 rows x 8 columns]

In [5]:
def createNewMpiCalculatedColumn(weightList, df, columnName):
    """Returns the percentage match of the poverty rankings between the two sets of data selected.
    :weightList: 
    :df: a dataframe with poverty data obtained by the survey data of world bank.
    :columnName: The column Whose weight is.
    :return: the column that has been manipulated.
    >>> primary_df = {"Country Numeric Code": ['360'], "Country Code": ['IDN'], "Country": ['Indonesia'], "World Region": ['East Asia and the Pacific'], "Survey": ['DHS'], "Year": ['2017'], "Multidimensional Poverty Index (MPI)": ['0.014011'],"Nutrition":['NaN'],"Child mortality":['1.460288'], "Years of schooling": ['1.551531'], "School attendance": ['0.69869'], "Cooking fuel": ['2.379687'],"Sanitation": ['2.181947'], "Drinking water": ['1.350654'], "Electricity": ['0.770641'],"Housing":['1.309179'],"Assets":['1.71485'],"Total number of indicators included":['9'],"New MPI":['NaN']}
    >>> df = {"Country": ['Indonesia'], "Population_Below_Poverty_National_Year": ['2018.0'], " Population_Below_Poverty_National": ['9.8'], " Population_Below_Poverty_Rural": ['1.7'], " Population_Below_Poverty_Urban": [’14.8'], " Poverty_Gap_Year ": ['2012.0'], " Reference_Year": ['2012.0']," World_Bank_Estimate":['29.5']," Income_Lowest_10":['3.7'], " Income_Lowest_20": ['8.8'], " Income_Highest_10": ['23.7'], " Baseline_Year": ['2014.0']," Recent_Year": ['2017.0'], " Annualized_Growth_Bottom_40": ['-0.1], " Annualized_Growth_Total_Population": ['1.9']," Mean_Consumption_Bottom_40_Baseline":['6.6']," Mean_Consumption_Bottom_40_Most_Recent":['6.6']," Mean_Consumption_Total_Population_Baseline ":['15.4']," Mean_Consumption_Total_Population_Most_Recent ":['16.2'], 10% Difference":['20.0']}    >>> getDifferenceInPovertyIn2DFs(primary_df, df, code_or_name_comparision)
    33.33
    >>> columnName: 'New MPI'
    """
    df[columnName] = ((df['Nutrition']*weightList[0]) + (df['Child mortality']*weightList[1]) + 
                      (df['Years of schooling']*weightList[2]) + (df['School attendance']*weightList[3]) +
                      (df['Cooking fuel']*weightList[4]) + (df['Sanitation']*weightList[5]) +
                      (df['Drinking water']*weightList[6]) + (df['Electricity']*weightList[7]) +
                      (df['Housing']*weightList[8]) + (df['Assets']*weightList[9]))/10
    return df[columnName]

In [6]:
def getDifferenceInPovertyIn2DFs(primary_df, df, code_or_name_comparision):
    """Returns the percentage match of the poverty rankings between the two sets of data selected.
    :primary_df: a dataframe consisting poverty data calculated using MPI.
    :df: a dataframe with poverty data obtained by the survey data of world bank.
    :code_or_name_comparision: a variable that stores the country code or the country name used for comparison.
    :return: a value indicating the percentage match 
    >>> primary_df = {"Country Numeric Code": ['360'], "Country Code": ['IDN'], "Country": ['Indonesia'], "World Region": ['East Asia and the Pacific'], "Survey": ['DHS'], "Year": ['2017'], "Multidimensional Poverty Index (MPI)": ['0.014011'],"Nutrition":['NaN'],"Child mortality":['1.460288'], "Years of schooling": ['1.551531'], "School attendance": ['0.69869'], "Cooking fuel": ['2.379687'],"Sanitation": ['2.181947'], "Drinking water": ['1.350654'], "Electricity": ['0.770641'],"Housing":['1.309179'],"Assets":['1.71485'],"Total number of indicators included":['9'],"New MPI":['NaN']}
    >>> code_or_name_comparision = 'Indonesia'
    >>> df = {"Country": ['Indonesia'], "Population_Below_Poverty_National_Year": ['2018.0'], " Population_Below_Poverty_National": ['9.8'], " Population_Below_Poverty_Rural": ['1.7'], " Population_Below_Poverty_Urban": [’14.8'], " Poverty_Gap_Year ": ['2012.0'], " Reference_Year": ['2012.0']," World_Bank_Estimate":['29.5']," Income_Lowest_10":['3.7'], " Income_Lowest_20": ['8.8'], " Income_Highest_10": ['23.7'], " Baseline_Year": ['2014.0']," Recent_Year": ['2017.0'], " Annualized_Growth_Bottom_40": ['-0.1], " Annualized_Growth_Total_Population": ['1.9']," Mean_Consumption_Bottom_40_Baseline":['6.6']," Mean_Consumption_Bottom_40_Most_Recent":['6.6']," Mean_Consumption_Total_Population_Baseline ":['15.4']," Mean_Consumption_Total_Population_Most_Recent ":['16.2'], 10% Difference":['20.0']}    >>> getDifferenceInPovertyIn2DFs(primary_df, df, code_or_name_comparision)
    33.33
    """
    print(primary_df, df, code_or_name_comparision)
    number_of_matches = 0
    for i in range(len(df)):
        for j in range(len(df)):
            if code_or_name_comparision == 'Code':
                if df['ISO'][j] == primary_df['Country Code'][i]:
                    print(df['Country'][i])
                    number_of_matches += 1
                    continue
            elif code_or_name_comparision == 'CountryName':
                if df['Country'][j] == primary_df['Country'][i]:
                    print(df['Country'][i])
                    number_of_matches += 1
                    continue
            
    return ((number_of_matches/len(df))*100)

The MPI list will be altered by changing the weight of the facators involved in the MPI. The list from the weighted factors of MPI that matches with the world bank poevrty data with highest accuracy will be taken as the new MPI based dataset.

In [7]:
weights = [0.06, 0.06, 0.16, 0.06, 0.06, 0.06, 0.16, 0.16, 0.16, 0.06]
new_column_name = 'New MPI'
new_df = mpi_df.loc[:, mpi_df.columns != "Indicator (s) missing"]
new_df[new_column_name] = createNewMpiCalculatedColumn(weights, new_df, new_column_name)
new_df = new_df.sort_values(by = 'New MPI')
new_df = new_df.reset_index(drop=True)
new_df


Country Numeric Code Country Code              Country  \
0                     688          SRB               Serbia   
1                      51          ARM              Armenia   
2                     398          KAZ           Kazakhstan   
3                     417          KGZ           Kyrgyzstan   
4                     275          PSE  Palestine, State of   
..                    ...          ...                  ...   
104                   484          MEX               Mexico   
105                   426          LSO              Lesotho   
106                   340          HND             Honduras   
107                   598          PNG     Papua New Guinea   
108                     4          AFG          Afghanistan   

                        World Region  Survey        Year  \
0            Europe and Central Asia     MICS       2019   
1            Europe and Central Asia      DHS  2015-2016   
2            Europe and Central Asia     MICS       2015   
3            Europe and Central Asia     MICS       2018   
4                        Arab States     MICS  2019-2020   
..                               ...      ...        ...   
104  Latin America and the Caribbean  ENSANUT       2016   
105               Sub-Saharan Africa     MICS       2018   
106  Latin America and the Caribbean      DHS  2011-2012   
107        East Asia and the Pacific      DHS  2016-2018   
108                       South Asia      DHS  2015-2016   

     Multidimensional Poverty Index (MPI)  Nutrition  Child mortality  \
0                                0.000433   0.023019         0.057378   
1                                0.000690   0.137095         0.000000   
2                                0.001611   0.451137         0.422889   
3                                0.001426   0.393031         0.159708   
4                                0.001980   0.496145         0.251218   
..                                    ...        ...              ...   
104                              0.025615   5.233283              NaN   
105                              0.084359   9.584669         1.484473   
106                              0.093057   9.923804         0.958571   
107                              0.263291        NaN         3.637692   
108                              0.271721        NaN         8.159285   

     Years of schooling  School attendance  Cooking fuel  Sanitation  \
0              0.054523           0.049654      0.100292    0.030504   
1              0.023653           0.128609      0.072571    0.190553   
2              0.001840           0.028248      0.030088    0.000000   
3              0.039468           0.113286      0.264173    0.052981   
4              0.043423           0.324919      0.033550    0.094957   
..                  ...                ...           ...         ...   
104            1.324077           0.778597      2.431594    2.161883   
105            5.519687           3.659291           NaN   14.803903   
106           10.231574           7.900505     19.154258   14.553490   
107           19.493614          27.990497     56.262477   50.799519   
108           31.381973          41.964449     49.259972   48.714027   

     Drinking water  Electricity    Housing     Assets  \
0          0.024812     0.007462   0.046886   0.015927   
1          0.038291     0.000000   0.023653   0.048983   
2          0.060462     0.001840   0.092562   0.001840   
3          0.096483     0.000000   0.036623   0.000000   
4          0.024368     0.000000   0.051991   0.012183   
..              ...          ...        ...        ...   
104        1.163711     0.139551   1.432236   1.070969   
105       11.618516    18.364160  15.892954  15.238950   
106        7.022158          NaN  18.460836   7.858169   
107       44.926346    55.261831  52.695402  49.699559   
108       30.120814    23.675098  54.479075  13.854264   

     Total number of indicators included    New MPI  
0                                 

In [8]:
#print(world_bank_df.loc[world_bank_df['Country'] == 'Indonesia'])
print(getDifferenceInPovertyIn2DFs(new_df.head(75), world_bank_df.head(75), 'CountryName'), "% match between MPI index and income based ratings")

    Country Numeric Code Country Code              Country  \
0                    688          SRB               Serbia   
1                     51          ARM              Armenia   
2                    398          KAZ           Kazakhstan   
3                    417          KGZ           Kyrgyzstan   
4                    275          PSE  Palestine, State of   
..                   ...          ...                  ...   
70                   834          TZA             Tanzania   
71                   646          RWA               Rwanda   
72                   694          SLE         Sierra Leone   
73                   800          UGA               Uganda   
74                   324          GIN               Guinea   

               World Region Survey        Year  \
0   Europe and Central Asia    MICS       2019   
1   Europe and Central Asia     DHS  2015-2016   
2   Europe and Central Asia    MICS       2015   
3   Europe and Central Asia    MICS       2018   
4    

The world bank poverty list is compared with the MPI data without changing the weights of the factor. As seen above, only 22% of the countries matched in the top 25 countries.

In [9]:
weights = [0.05, 0.02, 0.25, 0.03, 0.02, 0.03, 0.25, 0.25, 0.02, 0.03]
new_df['New MPI'] = createNewMpiCalculatedColumn(weights, new_df, new_column_name)
new_df = new_df.sort_values(by = 'New MPI')
new_df = new_df.reset_index(drop=True)

print("Weights being tested: ", weights)
print(getDifferenceInPovertyIn2DFs(new_df.head(75), world_bank_df.head(75), 'CountryName'), "% match between MPI index and income based ratings")
print()

weights = [0.45, 0.45, 0.01, 0.01, 0.01, 0.02, 0.01, 0.01, 0.02, 0.01]
new_df['New MPI'] = createNewMpiCalculatedColumn(weights, new_df, new_column_name)
new_df = new_df.sort_values(by = 'New MPI')
new_df = new_df.reset_index(drop=True)

print("Weights being tested: ", weights)
print(getDifferenceInPovertyIn2DFs(new_df.head(75), world_bank_df.head(75), 'CountryName'), "% match between MPI index and income based ratings")

Weights being tested:  [0.05, 0.02, 0.25, 0.03, 0.02, 0.03, 0.25, 0.25, 0.02, 0.03]
    Country Numeric Code Country Code              Country  \
0                    688          SRB               Serbia   
1                     51          ARM              Armenia   
2                    398          KAZ           Kazakhstan   
3                    275          PSE  Palestine, State of   
4                    417          KGZ           Kyrgyzstan   
..                   ...          ...                  ...   
70                   834          TZA             Tanzania   
71                   800          UGA               Uganda   
72                   694          SLE         Sierra Leone   
73                   646          RWA               Rwanda   
74                   624          GNB        Guinea-Bissau   

               World Region Survey        Year  \
0   Europe and Central Asia    MICS       2019   
1   Europe and Central Asia     DHS  2015-2016   
2   Europe and Centra

[75 rows x 19 columns]                Country  Population_Below_Poverty_National_Year  \
0      Slovak Republic                                  2017.0   
1              Morocco                                  2013.0   
2            St. Lucia                                  2016.0   
3   West Bank and Gaza                                  2016.0   
4   Russian Federation                                  2018.0   
..                 ...                                     ...   
70   Equatorial Guinea                                  2006.0   
71        Burkina Faso                                  2018.0   
72               Chile                                  2017.0   
73               Gabon                                  2017.0   
74             Iceland                                  2016.0   

    Population_Below_Poverty_National  Population_Below_Poverty_Rural  \
0                                12.2                             NaN   
1                                 4.8 

The weights nutrition=0.24, child mortality=0.24, years of schooling=0.24, school attendence=0.24, cooking fuel=0.05, sanitation=0.05, drinking water=0.01, electricity=0.01, housing=0.01, assets=0.01 gave the higest accuracy of 33.33% after testing it multiple combinations (only 3 of which is shown above). The country ranking list obtained with this weights will be used to test the hypothesis.

In [10]:
weights = [0.05, 0.02, 0.25, 0.03, 0.02, 0.03, 0.25, 0.25, 0.02, 0.03]
new_df['New MPI'] = createNewMpiCalculatedColumn(weights, new_df, new_column_name)
new_df = new_df.sort_values(by = 'New MPI')
new_df = new_df.reset_index(drop=True)

# Hypothesis 1

The MPI urban and MPI rural are accurate indicators of the poverty levels (can verify by comparing it with the % population listed below poverty level)

In [11]:
urban_sorted_df = national_mpi_df.sort_values(by = 'MPI Urban')
urban_sorted_df = urban_sorted_df.reset_index(drop=True)
print("Countries matching between the comparisons of MPI Urban levels and weighted MPI levels:")
urban_matching_percent = getDifferenceInPovertyIn2DFs(new_df.head(25), urban_sorted_df.head(25), 'Code')
print()
print(urban_matching_percent, "% match between MPI Urban levels and weighted MPI levels")

Countries matching between the comparisons of MPI Urban levels and weighted MPI levels:
    Country Numeric Code Country Code              Country  \
0                    688          SRB               Serbia   
1                     51          ARM              Armenia   
2                    398          KAZ           Kazakhstan   
3                    275          PSE  Palestine, State of   
4                    417          KGZ           Kyrgyzstan   
5                    268          GEO              Georgia   
6                    462          MDV             Maldives   
7                    400          JOR               Jordan   
8                    807          MKD      North Macedonia   
9                    776          TON                Tonga   
10                   764          THA             Thailand   
11                   780          TTO  Trinidad and Tobago   
12                   499          MNE           Montenegro   
13                     8          ALB       

In [12]:
rural_sorted_df = national_mpi_df.sort_values(by = 'MPI Rural')
rural_sorted_df = rural_sorted_df.reset_index(drop=True)
print("Countries matching between the comparisons of MPI Rural levels and weighted MPI levels:")
rural_matching_percent = getDifferenceInPovertyIn2DFs(new_df.head(25), rural_sorted_df.head(25), 'Code')
print()
print(rural_matching_percent, "% match between MPI Rural levels and weighted MPI levels")

Countries matching between the comparisons of MPI Rural levels and weighted MPI levels:
    Country Numeric Code Country Code              Country  \
0                    688          SRB               Serbia   
1                     51          ARM              Armenia   
2                    398          KAZ           Kazakhstan   
3                    275          PSE  Palestine, State of   
4                    417          KGZ           Kyrgyzstan   
5                    268          GEO              Georgia   
6                    462          MDV             Maldives   
7                    400          JOR               Jordan   
8                    807          MKD      North Macedonia   
9                    776          TON                Tonga   
10                   764          THA             Thailand   
11                   780          TTO  Trinidad and Tobago   
12                   499          MNE           Montenegro   
13                     8          ALB       

24                   6.02                            38.5   Code
Kazakhstan
Montenegro
Turkmenistan
Armenia
Bosnia and Herzegovina
Moldova, Republic of
Kyrgyzstan
Saint Lucia
Macedonia, The former Yugoslav Republic of
Thailand
Ukraine
Albania
Jamaica
Uzbekistan
Tunisia

60.0 % match between MPI Rural levels and weighted MPI levels


When comparing the MPI Urban level based poverty rankings to the newly weighted MPI level rankings, there was a 68% match in the rankings for the top 25 countries and comparing the MPI Rural level based poverty rankings, there was a 60% match in the rankings for the top 25 countries. Since there is no significant evidence to say that the MPI Urban and Rural indicators are not matching well with the actual MPI indicator, the hypothesis is accepted which means that the MPI urban and MPI rural are accurate indicators of the poverty levels.

# Hypothesis 2

The higher the difference between the income of the highest 10% and the bottom 10%, the higher the chances are for a country to fall under most poor countries

In [13]:
wb_df = world_bank_df
wb_df['10% Difference'] = wb_df['Income_Highest_10'] - wb_df['Income_Lowest_10']
wb_df = wb_df.sort_values(by='10% Difference', ascending=False)
wb_df = wb_df.reset_index(drop=True)
wb_df.head(10)

Country  Population_Below_Poverty_National_Year  \
0           Slovak Republic                                  2017.0   
1        Russian Federation                                  2018.0   
2                   Morocco                                  2013.0   
3            Cayman Islands                                     NaN   
4                  Mongolia                                  2018.0   
5        West Bank and Gaza                                  2016.0   
6                    Brazil                                     NaN   
7                    Belize                                     NaN   
8  St. Martin (French part)                                     NaN   
9                    Guinea                                  2018.0   

   Population_Below_Poverty_National  Population_Below_Poverty_Rural  \
0                               12.2                             NaN   
1                               12.9                             NaN   
2                                4.8                             1.4   
3                                NaN                            65.9   
4                               28.4                             0.9   
5                               29.2                             0.8   
6                                NaN                             4.4   
7                                NaN                            13.9   
8                                NaN                            23.4   
9                               43.7                            11.7   

   Population_Below_Poverty_Urban  Poverty_Gap_Year  Reference_Year  \
0                             NaN               NaN          2014.0   
1                             NaN               NaN          2017.0   
2                            15.0            2017.0          2015.0   
3                            82.9            2008.0          2008.0   
4                             7.3            2013.0          2014.0   
5                             4.5            2016.0          2015.0   
6                             9.1            2018.0          2018.0   
7                            27.7            1999.0          1999.0   
8                            42.8            1999.0          2016.0   
9                            25.2            1998.0          2010.0   

   World_Bank_Estimate  Income_Lowest_10  Income_Lowest_20  ...  \
0                 63.0               0.9               2.4  ...   
1                 56.3               1.3               3.9  ...   
2                 59.1               1.0               2.8  ...   
3                 56.2               1.2               3.3  ...   
4                 54.0               1.6               4.2  ...   
5                 57.1               1.0               2.9  ...   
6                 53.9               1.0               3.1  ...   
7                 53.3               0.9               3.2  ...   
8                 54.6               1.4               3.7  ...   
9                 50.7               1.6               4.5  ...   

   Income_Highest_10  Baseline_Year  Recent_Year  Annualized_Growth_Bottom_40  \
0               50.5            NaN          NaN                          NaN   
1               49.1            NaN          NaN                          NaN   
2               47.3            NaN          NaN                          NaN   
3               46.2            NaN          NaN                          NaN   
4               45.5            NaN          NaN                          NaN   
5               44.4         2011.0       2016.0                         -0.9   
6               42.5         2013.0       2018.0                         -0.4   
7               42.4            NaN          NaN                          NaN   
8               42.7            NaN          NaN                          NaN   
9               42.0            NaN          NaN                          NaN   

   Annualized_Growth_Total_Popula

In [14]:
print("Countries matching between the comparisons of difference between highest and lowest 10% income:")
world_bank_df = world_bank_df.sort_values(by = 'World_Bank_Estimate', ascending=False)
world_bank_df = world_bank_df.reset_index(drop=True)
matching_percent = getDifferenceInPovertyIn2DFs(wb_df.head(25), world_bank_df.head(25), 'CountryName')
print()
print(matching_percent, "% match between the difference between highest and lowest 10% income")

Countries matching between the comparisons of difference between highest and lowest 10% income:
                     Country  Population_Below_Poverty_National_Year  \
0            Slovak Republic                                  2017.0   
1         Russian Federation                                  2018.0   
2                    Morocco                                  2013.0   
3             Cayman Islands                                     NaN   
4                   Mongolia                                  2018.0   
5         West Bank and Gaza                                  2016.0   
6                     Brazil                                     NaN   
7                     Belize                                     NaN   
8   St. Martin (French part)                                     NaN   
9                     Guinea                                  2018.0   
10                  Botswana                                  2009.0   
11                    Angola            

[25 rows x 24 columns]                      Country  Population_Below_Poverty_National_Year  \
0            Slovak Republic                                  2017.0   
1                    Morocco                                  2013.0   
2                  St. Lucia                                  2016.0   
3         West Bank and Gaza                                  2016.0   
4         Russian Federation                                  2018.0   
5             Cayman Islands                                     NaN   
6   St. Martin (French part)                                     NaN   
7                   Mongolia                                  2018.0   
8                     Brazil                                     NaN   
9                     Belize                                     NaN   
10                  Botswana                                  2009.0   
11                     Haiti                                  2012.0   
12                    Angola             

After getting poverty rankings with the difference between the highest and lowest 10% income, it was compared to the weighted MPI list to check for similarity which gave a 84% match. The income difference list was sorted in reverse order so that the countries are listed according to their difference between the income of highest and lowest 10%. This is later compared to the MPI index list and got a 84% match, which allows us to conclude that 84% of the 25 countries with highest income difference appeared in the 25 countries list of the weighted MPI index. The higher accuracy allows us to accept the hypothesis which means that the higher the difference between the income of the highest 10% and the bottom 10%, the higher the chances are for a country to fall under most poor countries.

# Hypothesis 3

The higher the annualized growth of the bottom 40% of people in terms of income the higher the chances are for that country to push of the poverty zone

In [15]:
wb_df = world_bank_df
wb_df = wb_df.sort_values(by='Annualized_Growth_Bottom_40', ascending=False)
wb_df = wb_df.reset_index(drop=True)
wb_df.head(10)

Country  Population_Below_Poverty_National_Year  \
0       Portugal                                  2018.0   
1          Chile                                  2017.0   
2     Madagascar                                  2012.0   
3       Eswatini                                  2017.0   
4        Lao PDR                                  2018.0   
5     Luxembourg                                  2017.0   
6  Liechtenstein                                     NaN   
7       Paraguay                                  2018.0   
8     Mozambique                                  2014.0   
9           Cuba                                     NaN   

   Population_Below_Poverty_National  Population_Below_Poverty_Rural  \
0                               17.2                             3.1   
1                                8.6                             0.5   
2                               70.7                             0.0   
3                               58.9                             NaN   
4                               18.3                             0.0   
5                               18.3                             4.6   
6                                NaN                             0.3   
7                               24.2                             0.3   
8                               46.1                            13.8   
9                                NaN                             NaN   

   Population_Below_Poverty_Urban  Poverty_Gap_Year  Reference_Year  \
0                             6.3            2017.0          2018.0   
1                             5.4            2016.0          2016.0   
2                             0.2            2015.0          2016.0   
3                             NaN               NaN          2017.0   
4                             0.1            2011.0          2011.0   
5                             9.0            2017.0          2012.0   
6                             0.4            2017.0          2017.0   
7                             0.6            2017.0          2017.0   
8                            30.3            2015.0          2010.0   
9                             NaN               NaN          2017.0   

   World_Bank_Estimate  Income_Lowest_10  Income_Lowest_20  ...  \
0                 37.5               2.9               7.1  ...   
1                 38.5               2.7               6.5  ...   
2                 31.3               3.4               8.3  ...   
3                 30.4               3.0               8.1  ...   
4                 31.8               3.1               7.9  ...   
5                 42.6               2.2               5.7  ...   
6                 34.9               2.4               6.5  ...   
7                 33.8               2.7               7.4  ...   
8                 32.8               3.5               8.3  ...   
9                 30.4               2.7               7.4  ...   

   Income_Highest_10  Baseline_Year  Recent_Year  Annualized_Growth_Bottom_40  \
0               29.9         2012.0       2017.0                         10.0   
1               29.4         2013.0       2016.0                          8.4   
2               25.2         2015.0       2016.0                          8.3   
3               22.5         2012.0       2017.0                          8.0   
4               24.8         2012.0       2017.0                          7.9   
5               33.5         2012.0       2017.0                          7.1   
6               25.8         2012.0       2017.0                          6.3   
7               26.7         2012.0       2017.0                          6.0   
8               26.4         2009.0       2015.0                          5.7   
9               22.8         2012.0       2017.0                          5.5   

   Annualized_Growth_Total_Population  Mean_Consumption_Bottom_40_Baseline  \
0                                10.0                            

In [16]:
print("Countries matching after the comparisons of Annualized Growth of Bottom 40% income bracket people:")
world_bank_df = world_bank_df.sort_values(by = 'World_Bank_Estimate')
world_bank_df = world_bank_df.reset_index(drop=True)
matching_percent = getDifferenceInPovertyIn2DFs(wb_df.head(25), new_df.head(25), 'CountryName')
print()
print(matching_percent, "% match after the comparisons of Annualized Growth of Bottom 40% income bracket people")

Countries matching after the comparisons of Annualized Growth of Bottom 40% income bracket people:
                           Country  Population_Below_Poverty_National_Year  \
0                         Portugal                                  2018.0   
1                            Chile                                  2017.0   
2                       Madagascar                                  2012.0   
3                         Eswatini                                  2017.0   
4                          Lao PDR                                  2018.0   
5                       Luxembourg                                  2017.0   
6                    Liechtenstein                                     NaN   
7                         Paraguay                                  2018.0   
8                       Mozambique                                  2014.0   
9                             Cuba                                     NaN   
10             Trinidad and Tobago         

Tonga
Thailand
Montenegro
Tajikistan

16.0 % match after the comparisons of Annualized Growth of Bottom 40% income bracket people


The country poverty list based on the highest annualized grwoth of bottom 40% income bracket people was prepared and compared with the weighted MPI list and only 16% of countries came to be common, which means that only 12% of the top 25 countries in the annualized grwoth list was found in the top 25 of the MPI list. This is not a strong enough evidence to accept the hypothesis. Hence, the hypothesis is rejected meaning the higher the annualized growth of the bottom 40% of people in terms of income does not mean that chances for that country to push of the poverty zone is high.   